kernel/mem_domain.c

```diff
@@ -168,6 +168,11 @@ int k_mem_domain_add_partition(struct k_mem_domain *domain,
 		goto out;
 	}
 
+#ifdef Z_LIBC_PARTITION_EXISTS
+	if (part->size == 0 && part == &z_libc_partition)
+		goto out;
```

Comment:
I think this should return an error instead.

Reply:
The reason for this change is to make having an empty libc partition *not* be an error. Without this change, it would be an error to have a zero sized libc partition already.

Predicting when the libc partition will be empty is going to be tricky when malloc is no longer always included. The malloc static variables live in the libc partition but they only land in the binary if there is a call to malloc somewhere in the application. And that can depend on a wide range of subsystem versions and configuration options.